valkey_glide_otel.c

```diff
@@ -0,0 +1,265 @@
+        long flush_ms = Z_LVAL_P(flush_val);
+        if (flush_ms <= 0) {
+            VALKEY_LOG_ERROR("otel_config", "Flush interval must be a positive integer");
+            return 0;
+        }
+        otel_config->config->flush_interval_ms = (uint32_t) flush_ms;
+    }
+
+    /* Validate at least one of traces or metrics is configured (like Java/Go) */
+    if (!otel_config->config->traces && !otel_config->config->metrics) {
+        VALKEY_LOG_ERROR("otel_config", "At least one of traces or metrics must be configured");
+        return 0;
+    }
+
+    return 1;
+}
+
+/**
+ * Cleanup OTEL configuration
+ */
+void cleanup_otel_config(valkey_glide_otel_config_t* otel_config) {
+    if (otel_config->traces_config) {
+        if (otel_config->traces_config->endpoint) {
+            efree((void*) otel_config->traces_config->endpoint);
+            otel_config->traces_config->endpoint = NULL;
+        }
+        efree(otel_config->traces_config);
+        otel_config->traces_config = NULL;
+    }
+
+    if (otel_config->metrics_config) {
+        if (otel_config->metrics_config->endpoint) {
+            efree((void*) otel_config->metrics_config->endpoint);
+            otel_config->metrics_config->endpoint = NULL;
+        }
+        efree(otel_config->metrics_config);
+        otel_config->metrics_config = NULL;
+    }
+
+    if (otel_config->config) {
+        /* Reset pointers to avoid double-free */
+        otel_config->config->traces  = NULL;
+        otel_config->config->metrics = NULL;
+        efree(otel_config->config);
+        otel_config->config = NULL;
+    }
+
+    otel_config->enabled                   = false;
+    otel_config->current_sample_percentage = 0;
+}
+
+/* Public API function implementations */
+
+bool valkey_glide_otel_is_initialized(void) {
+    return g_otel_config.enabled;
+}
+
+int32_t valkey_glide_otel_get_sample_percentage(void) {
+    if (!g_otel_config.enabled || !g_otel_config.traces_config) {
+        return -1;  // Not initialized or no traces config
+    }
+    return (int32_t) g_otel_config.traces_config->sample_percentage;
+}
+
+bool valkey_glide_otel_set_sample_percentage(int32_t percentage) {
+    if (!g_otel_config.enabled || !g_otel_config.traces_config) {
+        return false;  // Not initialized or no traces config
+    }
+
+    if (percentage < 0 || percentage > 100) {
+        return false;  // Invalid percentage
+    }
+
+    g_otel_config.traces_config->sample_percentage = (uint32_t) percentage;
+    return true;
+}
+
+bool valkey_glide_otel_should_sample(void) {
+    int32_t percentage = valkey_glide_otel_get_sample_percentage();
+    if (percentage < 0) {
+        return false;  // Not initialized or no traces config
+    }
+
+    if (percentage == 0) {
+        return false;  // 0% sampling
+    }
+
+    if (percentage == 100) {
+        return true;  // 100% sampling
+    }
+
+    // Random sampling based on percentage
+    return (rand() % 100) < percentage;
+}
```

Comment:
Nit. Simplification if you want.

```suggestion
bool valkey_glide_otel_should_sample(void) {
    int32_t percentage = valkey_glide_otel_get_sample_percentage();
    return percentage > 0 && (percentage == 100 || (rand() % 100) < percentage);
}
```

Reply:
Updated.